core/src/main/java/io/opentelemetry/android/export/SpanDataModifier.java

```diff
@@ -132,7 +132,18 @@ public SpanExporter build() {
         if (!spanAttributeReplacements.isEmpty()) {
             modifier =
                     new AttributeModifyingSpanExporter(
-                            delegate, new HashMap<>(spanAttributeReplacements));
+                            delegate,
+                            (attributeKey, value) -> {
+                                //noinspection unchecked
+                                Function<Object, Object> function =
+                                        (Function<Object, Object>)
+                                                spanAttributeReplacements.get(attributeKey);
+                                if (function != null) {
+                                    return function.apply(value);
+                                } else {
+                                    return value;
```

Comment:
nit: Prefer omitting the redundant else (and yeah, if it was kotlin it would be even cleaner to return from the `if` I suppose).